dbms/src/Common/ColumnsHashing.h

```diff
@@ -90,19 +91,10 @@ struct HashMethodString
         return StringRef(chars + offsets[row - 1], offsets[row] - offsets[row - 1] - 1);
     }
 
-    static StringRef getValueRef(const Value & value) { return StringRef(value.getFirst().data, value.getFirst().size); }
+    static StringRef getValueRef(const Value & value) { return value.first; }
 
 protected:
     friend class columns_hashing_impl::HashMethodBase<Self, Value, Mapped, use_cache>;
-
-    static ALWAYS_INLINE void onNewKey([[maybe_unused]] StringRef & key, [[maybe_unused]] Arena & pool)
-    {
```

Comment:
If we're getting rid of onNewKey method, do we have to also change onExistingKey? It is non-empty for HashMethodSerialize. We don't work with serialized keys for now, but if we have a common API for placing keys to Arena, it's better to have it work uniformly for all cases.

Reply:
onExistingKey is different from onNewKey. It's for rollback the storage of the getKey methond. Though they sound like a pair.